contributors/design-proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,405 @@
+# Alternate representations of API resources
+
+## Abstract
+
+Naive clients benefit from allowing the server to returning resource information in a form
+that is easy to represent or is more efficient when dealing with resources in bulk. It
+should be possible to ask an API server to return a representation of one or more resources
+of the same type in a way useful for:
+
+* Retrieving a subset of object metadata in a list or watch of a resource, such as the
+  metadata needed by the generic Garbage Collector or the Namespace Lifecycle Controller
+* Dealing with generic operations like `Scale` correctly from a client across multiple API
+  groups, versions, or servers
+* Return a simple tabular representation of an object or list of objects for naive
```

Comment:
None of our fields are simple.  If you look at get describers, the location of that code is arbitrarily in kubectl, but it really belongs on the server.  You don't *have* to use that format, but most clients want it.  OpenAPI isn't enough for the use cases we have today, but yes, simple implementations could exist.  But having them on the client isn't really helping.